File: uniquePaths.py
# https://leetcode.com/problems/unique-paths/ 

class Solution(object):
	def uniquePaths(self, m, n):
		dp = [[0 for _ in range(n)] for _ in range(m)]
		dp[0][0] = 1
		for i in range(m):
			for j in range(n):
				if i - 1 >= 0:
					dp[i][j] += dp[i - 1][j]
                    
				if j - 1 >= 0:
					dp[i][j] += dp[i][j - 1]

		return dp[-1][-1]

File: uniquePaths.java
// https://leetcode.com/problems/unique-paths/ 

public class Solution {
	public int uniquePaths(int m, int n) {
		int[][] paths = new int[m][n];
		paths[0][0] = 1;
        
		for (int i = 0; i < m; i++) {
			for (int j = 0; j < n; j++) {
				if (i - 1 >= 0) {
					paths[i][j] += paths[i - 1][j];
				}

				if (j - 1 >= 0) {
					paths[i][j] += paths[i][j - 1];
				}
		    }
		}

		return paths[m - 1][n - 1];
	}
}
